src/elements/element.line.js

```diff
@@ -25,84 +23,141 @@ defaults._set('global', {
 	}
 });
 
+function startAtGap(points, spanGaps) {
+	let closePath = true;
+	let previous = points.length && points[0]._view;
+	let index, view, cloned;
+	for (index = 1; index < points.length; ++index) {
+		// If the line has an open path, shift the point array
+		view = points[index]._view;
+		if (!view.skip && previous.skip) {
+			points = points.slice(index).concat(points.slice(0, index));
+			closePath = spanGaps;
+			cloned = true;
+			break;
+		}
+		previous = view;
+	}
+	// If the line has a close path, add the first point again
+	if (closePath) {
+		if (!cloned) {
+			points = points.slice();
+		}
+		points.push(points[0]);
+	}
+	points.closePath = closePath;
+	return points;
+}
+
+function setStyle(ctx, vm) {
+	ctx.lineCap = vm.borderCapStyle;
+	// IE 9 and 10 do not support line dash
```

Comment:
We don't support IE 9 and 10, so potentially this could be removed. Though I understand you're just moving existing code, so am also fine if you want to leave as is

Reply:
removed the test